packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+module.exports = compile
+
+function compile (node) {
+  if (node === null || typeof node === 'number' || typeof node === 'boolean' || typeof node === 'string') {
+    return JSON.stringify(node)
+  }
+
+  const [type, value] = Object.entries(node)[0]
+
+  if (type === 'not') {
+    return `!(${compile(value)})`
+  } else if (type === 'isEmpty') {
+    return `(${compile(value)}).length === 0`
+  } else if (type === 'isDefined') {
+    return `(() => {
+        try {
+          ${value}
+          return true
+        } catch {
+          return false
+        }
+      })()`
+  } else if (type === 'instanceof') {
+    return `${compile(value[0])} instanceof ${value[1]}`
+  } else if (type === 'len' || type === 'count') {
+    return `(${compile(value)}).length`
+  } else if (type === 'ref') {
+    if (value === '@it') {
+      return 'it'
+    } else if (value === '@key') {
+      return 'key'
+    } else if (value === '@value') {
+      return 'value'
+    } else {
+      return value
+    }
+  } else if (Array.isArray(value)) {
+    const args = value.map(compile)
+    switch (type) {
+      case 'eq': return `${args[0]} === ${args[1]}`
+      case 'ne': return `${args[0]} !== ${args[1]}`
+      case 'gt': return `${args[0]} > ${args[1]}`
+      case 'ge': return `${args[0]} >= ${args[1]}`
+      case 'lt': return `${args[0]} < ${args[1]}`
+      case 'le': return `${args[0]} <= ${args[1]}`
+      case 'any': return iterateOn('some', ...args)
+      case 'all': return iterateOn('every', ...args)
+      case 'and': return `(${args.join(') && (')})`
+      case 'or': return `(${args.join(') || (')})`
+      case 'startsWith': return `${args[0]}.startsWith(${args[1]})`
```

Comment:
should we put `()` round ${args[0]} here?

Reply:
Fixed 👍